v2/internal/pkg/operator/common.go

```diff
@@ -161,8 +162,22 @@ func (o OperatorCollector) prepareD2MCopyBatch(images map[string][]v2alpha1.Rela
 			}
 
 			// add the tag for src and dest
-			switch {
 			// applies only to catalogs
+			switch {
+			// OCPBUGS-52470
+			case img.Type == v2alpha1.TypeOperatorCatalog && len(img.TargetTag) == 0 && imgSpec.IsImageByDigestOnly():
+				// find the tag for for the catalog that was mirrored via digest
+				imageIndexDir := filepath.Join(o.Opts.Global.WorkingDir, operatorCatalogsDir, imgSpec.ComponentName(), imgSpec.Digest)
+				filteredCatalogsDir := filepath.Join(imageIndexDir, operatorCatalogFilteredDir)
+				entries, err := os.ReadDir(filteredCatalogsDir)
+				if err != nil {
+					return result, fmt.Errorf("could not find directory filtered catalog image %w", err)
+				}
+				if len(entries) > 1 {
+					return result, fmt.Errorf("the directory structure for digest based catalogs is incorrect")
+				}
```

Comment:
Hi @lmzuccarelli 
You're right !
This could happen if several mirrorings for the same catalog but with  different filterings have been performed...

`o.Config.Mirror.Operators` could give us an access to the filter, if that could help... It would need some matching logic to find the right operator though.

Another possibility is to use `RebuiltTag` from `v2alpha1.RelatedImage` ... 
Granted, this field doesn't contain what we need today on D2M in the case of mirroring catalogs by digest, but we could also prepare it in order to have it when in `prepareD2MCopyBatch` .

Reply:
Hi @lmzuccarelli 
Since this case can happen (multiple digests for the same catalog). what did you think of the proposals above?